### PR TITLE
Sitton simple drivetrain

### PR DESCRIPTION
**Pull request by tal-sitton**

### CODE REVIEW

File: src/main/java/frc/robot/drivetrains/simpleTankDrivetrain/Robot.java [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+import edu.wpi.first.networktables.NetworkTable;
+import edu.wpi.first.wpilibj.TimedRobot;
+import edu.wpi.first.wpilibj.motorcontrol.MotorControllerGroup;
+import edu.wpi.first.wpilibj2.command.CommandScheduler;
+
+public class Robot extends TimedRobot {
+
+    /**
+     * <p> A namespace is an object that holds values on a {@link NetworkTable}.</p>
+     * This is the main namespace which should host the main values and commands which don't belong to a single subsystem
```
Comment (OfriRosenbaum) [inline src/main/java/frc/robot/drivetrains/simpleTankDrivetrain/Robot.java]: change `which don't belong` to `that don't belong`

File: src/main/java/frc/robot/drivetrains/simpleTankDrivetrain/Robot.java [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+     * or command.
+     */
+    private final RootNamespace robotNamespace = new RootNamespace("robot");
```
Comment (OfriRosenbaum) [inline src/main/java/frc/robot/drivetrains/simpleTankDrivetrain/Robot.java]: maybe you should update this namespace?

File: src/main/java/frc/robot/drivetrains/simpleTankDrivetrain/OI.java [hunk 3]
```diff
@@ -0,0 +1,23 @@
+package frc.robot.drivetrains.simpleTankDrivetrain;
+
+import edu.wpi.first.wpilibj.Joystick;
+
+/**
+ * Handles the communication between the user and the robot.
+ */
+public class OI /*GEVALD*/ {
+
+    private final Joystick leftJoystick = new Joystick(0);
```
Comment (Yoel-Perman-Brilliant) [inline src/main/java/frc/robot/drivetrains/simpleTankDrivetrain/OI.java]: In the latest RapidReact pr, the joysticks are static. These should probably be static too.

File: src/main/java/frc/robot/drivetrains/simpleTankDrivetrain/Robot.java [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+
+    /**
+     * This simple drivetrain has only 4 talons, 2 for each side. <br>
+     * The <b>ONLY</b> thing you need to change is the ports in the {@link RobotMap}
```
Comment (OfriRosenbaum) [inline src/main/java/frc/robot/drivetrains/simpleTankDrivetrain/Robot.java]: This looks like a sentence to me, but where is the period?

File: src/main/java/frc/robot/drivetrains/simpleTankDrivetrain/Robot.java [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+
+    @Override
+    public void autonomousPeriodic() {
+    }
+
+    @Override
+    public void teleopInit() {
+        DriveArcade driveArcade = new DriveArcade(drivetrain, oi::getLeftX, oi::getRightY);
+        drivetrain.setDefaultCommand(driveArcade);
+
```
Comment (OfriRosenbaum) [inline src/main/java/frc/robot/drivetrains/simpleTankDrivetrain/Robot.java]: why blank line

File: src/main/java/frc/robot/drivetrains/simpleTankDrivetrain/OI.java [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+    private final Joystick rightJoystick = new Joystick(1);
+
+    public OI() {
+    }
+
+    public double getLeftX() {
```
Comment (Yoel-Perman-Brilliant) [inline src/main/java/frc/robot/drivetrains/simpleTankDrivetrain/OI.java]: add `getRightX()` and `getLeftY()`. Even if they're not used, a team that copy-pastes the code might want to use the joysticks differently

**Review by OfriRosenbaum — CHANGES_REQUESTED** [review]

File: src/main/java/frc/robot/drivetrains/simpleTankDrivetrain/Robot.java [hunk 7]
```diff
@@ -0,0 +1,94 @@
+// Copyright (c) FIRST and other WPILib contributors.
+// Open Source Software; you can modify and/or share it under the terms of
+// the WPILib BSD license file in the root directory of this project.
+
+package frc.robot.drivetrains.simpleTankDrivetrain;
```
Comment (Yoel-Perman-Brilliant) [inline src/main/java/frc/robot/drivetrains/simpleTankDrivetrain/Robot.java]: package names don't have a strict naming convention, but they're usually completely in lowercase (e.g. `genericsubsytem`). So change this to `simpletankdrivetrain`.

File: src/main/java/frc/robot/drivetrains/simpleTankDrivetrain/Robot.java (hunk 4 above)
Comment (Yoel-Perman-Brilliant) [inline src/main/java/frc/robot/drivetrains/simpleTankDrivetrain/Robot.java]: add `.` add the end

File: src/main/java/frc/robot/drivetrains/simpleTankDrivetrain/RobotMap.java [hunk 8]
```diff
@@ -0,0 +1,24 @@
+package frc.robot.drivetrains.simpleTankDrivetrain;
+
+public class RobotMap {
+
+    public interface CAN {
+
+        int LEFT_TALON_1 = 0;
```
Comment (Yoel-Perman-Brilliant) [inline src/main/java/frc/robot/drivetrains/simpleTankDrivetrain/RobotMap.java]: change to `DRIVETRAIN_LEFT_TALON_1` (again, consider teams copy-pasting)

File: src/main/java/frc/robot/drivetrains/tankDrivetrain/Drivetrain.java [hunk 9]
```diff
@@ -1,4 +1,4 @@
-package frc.robot.drivetrains.tankdrivetrain;
```
Comment (Yoel-Perman-Brilliant) [inline src/main/java/frc/robot/drivetrains/tankDrivetrain/Drivetrain.java]: change to `advancedtankdrivetrain`

**Review by Yoel-Perman-Brilliant — CHANGES_REQUESTED** [review]

File: src/main/java/frc/robot/drivetrains/simpleTankDrivetrain/Robot.java (hunk 7 above)
Comment (OfriRosenbaum) [reply src/main/java/frc/robot/drivetrains/simpleTankDrivetrain/Robot.java]: A-I/a4

Comment (Yoel-Perman-Brilliant) [reply src/main/java/frc/robot/drivetrains/simpleTankDrivetrain/Robot.java]: thank you

File: src/main/java/frc/robot/drivetrains/simpleTankDrivetrain/Robot.java (hunk 4 above)
Comment (Yoel-Perman-Brilliant) [reply src/main/java/frc/robot/drivetrains/simpleTankDrivetrain/Robot.java]: also, consider they might also use different motorcontrollers, so they would also have to change those

File: src/main/java/frc/robot/drivetrains/simpletankdrivetrain/Robot.java [hunk 10]
```diff
@@ -24,7 +25,8 @@
 
     /**
      * This simple drivetrain has only 4 talons, 2 for each side. <br>
-     * The <b>ONLY</b> thing you need to change is the ports in the {@link RobotMap}.
+     * The <b>ONLY</b> thing you need to change is the ports in the {@link RobotMap},
+     * and maybe the type of your {@link MotorController}
```
Comment (OfriRosenbaum) [inline src/main/java/frc/robot/drivetrains/simpletankdrivetrain/Robot.java]: add a period

File: src/main/java/frc/robot/drivetrains/simpletankdrivetrain/Robot.java [hunk 11]
```diff
@@ -0,0 +1,96 @@
...
+    public void disabledInit() {
+    }
+
+    @Override
+    public void disabledPeriodic() {
+    }
+
+    @Override
+    public void autonomousInit() {
+
```
Comment (OfriRosenbaum) [inline src/main/java/frc/robot/drivetrains/simpletankdrivetrain/Robot.java]: remove blank line

**Review by OfriRosenbaum — CHANGES_REQUESTED** [review]

File: src/main/java/frc/robot/drivetrains/simpletankdrivetrain/Robot.java [hunk 12]
```diff
@@ -0,0 +1,95 @@
...
+    /**
+     * <p> A namespace is an object that holds values on a {@link NetworkTable}.</p>
+     * This is the main namespace which should host the main values and commands that don't belong to a single subsystem
+     * or command.
+     */
+    private final RootNamespace robotNamespace = new RootNamespace("robot");
+
+    /**
+     * This simple drivetrain has only 4 talons, 2 for each side. <br>
+     * The <b>ONLY</b> thing you need to change is the ports in the {@link RobotMap},
```
Comment (OfriRosenbaum) [inline src/main/java/frc/robot/drivetrains/simpletankdrivetrain/Robot.java]: change it to `the only things you may have to change are the ports... and the type...`

**Review by OfriRosenbaum — CHANGES_REQUESTED** [review]

**Review by OfriRosenbaum — APPROVED** [review]

**Pull request merged**